### PR TITLE
Update SonarLint config

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**

### CODE REVIEW

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 1]
```diff
@@ -28,6 +28,7 @@
     <Rule Id="S1125" Action="Info" />
     <Rule Id="S1128" Action="Warning" />
     <Rule Id="S113" Action="Info" />
+    <Rule Id="S1133" Action="Info" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:31]: S1133 - Deprecated code should be removed

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 2]
```diff
@@ -92,12 +93,14 @@
     <Rule Id="S2123" Action="Warning" />
     <Rule Id="S2148" Action="Info" />
     <Rule Id="S2156" Action="Info" />
+    <Rule Id="S2166" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:96]: S2166 - Classes named like "Exception" should extend "Exception" or a subclass

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 3]
```diff
@@ -92,12 +93,14 @@
...
     <Rule Id="S2178" Action="Warning" />
     <Rule Id="S2183" Action="Info" />
     <Rule Id="S2184" Action="Info" />
     <Rule Id="S2187" Action="Warning" />
     <Rule Id="S2190" Action="Warning" />
     <Rule Id="S2197" Action="Warning" />
+    <Rule Id="S2198" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:103]: S2198 - Silly mathematical comparisons should not be made

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 4]
```diff
@@ -172,6 +175,7 @@
     <Rule Id="S3052" Action="Info" />
     <Rule Id="S3059" Action="None" />
     <Rule Id="S3060" Action="Warning" />
+    <Rule Id="S3063" Action="Warning" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:178]: S3063 -  "StringBuilder" data should be used

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 5]
```diff
@@ -251,6 +255,7 @@
     <Rule Id="S3875" Action="Warning" />
     <Rule Id="S3876" Action="Info" />
     <Rule Id="S3877" Action="Warning" />
+    <Rule Id="S3878" Action="Info" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:258]: S3878 - Arrays should not be created for params parameters

File: analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset [hunk 6]
```diff
@@ -354,6 +359,7 @@
     <Rule Id="S4583" Action="Warning" />
     <Rule Id="S4586" Action="Warning" />
     <Rule Id="S4635" Action="Warning" />
+    <Rule Id="S4663" Action="Info" />
```
Comment (martin-strecker-sonarsource) [inline analyzers/.sonarlint/sonaranalyzer-dotnetCSharp.ruleset:362]: S4663 - Comments should not be empty

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

Please do the same on autoscan too, while we have the context

**Pull request merged**